plugins/linux-v4l2/v4l2-output.c

```diff
@@ -220,18 +209,63 @@ static bool virtualcam_start(void *data)
 			     "v4l2-output: A format truncation may have occurred."
 			     " This can be ignored since it is quite improbable.");
 
-		if (try_connect(vcam, device)) {
-			success = true;
-			break;
+		int fd = open(device, O_RDWR);
+		if (fd < 0)
+			continue;
+
+		if (ioctl(fd, VIDIOC_QUERYCAP, &v4l2cap) < 0)
+			goto try_next;
+
+		blog(LOG_DEBUG, "v4l2-output: device: '%s'", v4l2cap.card);
+
+		if (name && strncmp((const char *)v4l2cap.card, name,
+				    strlen(name)) != 0)
+			goto try_next;
+
+		if (!try_configure(data, fd)) {
+		try_next:
+			close(fd);
+			continue;
 		}
+
+		success = true;
+		break;
 	}
 
 	while (n--)
 		free(list[n]);
 	free(list);
 
-	if (!success)
+	return success;
+}
+
+static bool virtualcam_start(void *data)
+{
+	struct virtualcam_data *vcam = (struct virtualcam_data *)data;
+
+	if (!loopback_module_loaded()) {
+		if (loopback_module_load() != 0) {
+			return false;
+		}
+	}
+
+	bool success = open_loopback_card(data, OBS_V4L2_CARD_LABEL);
+
+	if (!success) {
+		loopback_module_add_card();
```

Comment:
Can we not get the device number back out of this operation?

Reply:
Added a TODO: need to refactor `run_command` to return the cmd output, as one way of getting the new device id.